ndcube/asdf/resources/schemas/ndcollection-1.0.0.yaml

```diff
@@ -0,0 +1,30 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "asdf://sunpy.org/ndcube/schemas/ndcollection-1.0.0"
+
+title:
+  Represents the ndcube.ndcollection.NDCollection object
+
+description:
+  Represents the ndcube ndcube.ndcollection.NDCollection object
+
+type: object
+properties:
+  keys:
+    type: array
+  value:
+    type: array
```

Comment:
I'm not a massive fan of this layout. I'd rather it were something like:
```
items:
  type: object
  properties:
    *: type:object
```
i.e. where we allow any keys but set their value.

@braingram is that a thing asdf can handle? jsonschema seems to have a `patternProperties` which lets you define a regex for keys (https://tour.json-schema.org/content/03-Objects/01-Pattern-Properties).